serverkeyapi/serverkeyapi_test.go

```diff
@@ -0,0 +1,313 @@
+package serverkeyapi
+
+import (
+	"bytes"
+	"context"
+	"crypto/ed25519"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/matrix-org/dendrite/internal/caching"
+	"github.com/matrix-org/dendrite/internal/config"
+	"github.com/matrix-org/dendrite/serverkeyapi/api"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+type server struct {
+	name      gomatrixserverlib.ServerName
+	validity  time.Duration
```

Comment:
Is this "valid for duration from now" or "valid for duration from key creation"?

Reply:
Well, it's effectively just writing to `config.Matrix.KeyValidityPeriod`.

The behaviour around `/_matrix/key/v2/server` has always been that we will return our key with a validity of `time.Now() + validity`. The remote server can re-request our key as often as they want but will normally do so when they reach the validity.

Reply:
So this is the "valid for duration from now"? Please comment to that effect, thanks!